### PR TITLE
#2359 - Update Vuetify and create virtual component for v-divider

### PR DESCRIPTION
**Pull request by ann-aot**
- Updated to latest vuetify version `[3.4.2](https://www.npmjs.com/package/vuetify/v/3.4.2)` 
- Update typescript pacakege to `4.9.5` from `4.5.5`. I didn't install the latest version of the typescript `[5.3.2](https://www.npmjs.com/package/typescript/v/5.3.2)`, because vue3-jest lastest version is `27.0.0-alpha.1`, which need bable-jest with version 27.x and ts-jest@"27. x" and jest@27.x, which need typescript@">=3.8 <5.0", that's the reason, I installed the latest 4 versions of typescript. Added some screenshots below,
image
image
image
- created 2 create virtual components for Vdivider and added to the existing dividers
- data table import is now from vuetify components,  (it was earlier from vuetify labs)
**New Updates**
- With the new data table import, there were some issues, fixed the issues as part of the PR

> - `raw` and `columns` is no longer available in `item` in the data table

- `getAllInProgressApplications` was using `InProgressApplicationOfferingChangesAPIOutDTO` which is an institution DTO on the web side, updated it to `AllInProgressApplicationOfferingChangesAPIOutDTO` to sync it up with API.
- `active-color` is deprecated in `v-list`, because of this there were some 2000+ warnings on institution loading, replaced it with color
image

### CODE REVIEW

File: sources/packages/web/src/components/aest/institution/ApplicationOfferingChangeDetailsHeader.vue [hunk 1]
```diff
@@ -9,12 +9,9 @@
             : '-'
         "
       />
-      <v-divider
+      <v-divider-vertical-opaque
```
Comment (dheepak-aot) [inline sources/packages/web/src/components/aest/institution/ApplicationOfferingChangeDetailsHeader.vue:12]: 👍

File: sources/packages/web/src/plugins/vuetify.ts [hunk 2]
```diff
@@ -8,15 +8,40 @@ import { VDataTableServer, VDataTable } from "vuetify/labs/VDataTable";
...
   },
+  aliases: {
+    VDividerOpaque: VDivider,
+    VDividerInsetOpaque: VDivider,
+    VDividerVerticalOpaque: VDivider,
+  },
   directives,
   default: {},
+  defaults: {
+    VDividerOpaque: {
```
Comment (dheepak-aot) [inline sources/packages/web/src/plugins/vuetify.ts:27]: 👍

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 3]
```diff
@@ -35,13 +35,10 @@
...
                 </v-list-item>
-                <v-divider
-                  class="border-opacity-100"
-                  :thickness="2"
+                <v-divider-inset-opaque
                   v-if="index < items.length - 1"
                   :key="index"
-                  inset
-                ></v-divider>
+                ></v-divider-inset-opaque>
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:41]: minor. the element can be terminated inline. I know it is from existing code.

```html
<v-divider-inset-opaque
                  v-if="index < items.length - 1"
                  :key="index"
                />
```

**Review by dheepak-aot — APPROVED** [review]
Thanks for the effort to upgrade typescript as well. 👍
Had one very minor comment and it is optional.

**Review by andrepestana-aot — APPROVED** [review]
Looks good! Just a minor comment from Dheepak that can be applied to some other places.

**Review by andrewsignori-aot — APPROVED** [review]
The changes look good and thanks for upgrading Typescript also 👍 
Should we have a ticket to migrate Vue CLI that it is now in maintenance mode?
https://github.com/vuejs/vue-cli#%EF%B8%8F-status
image

File: sources/packages/web/src/plugins/vuetify.ts [hunk 4]
```diff
@@ -3,12 +3,10 @@ import "@fortawesome/fontawesome-free/css/all.css";
 import "vuetify/lib/styles/main.sass";
 import { createVuetify } from "vuetify";
 import * as components from "vuetify/lib/components";
-// Todo: remove import from vuetify/labs/ and its corresponding logics when the components are available in vetify/lib/components.
-import { VDataTableServer, VDataTable } from "vuetify/labs/VDataTable";
 import * as directives from "vuetify/lib/directives";
 import { aliases, fa } from "vuetify/iconsets/fa";
 import { mdi } from "vuetify/iconsets/mdi";
-import { VDivider } from "vuetify/lib/components";
+import { VDivider, VDataTableServer, VDataTable } from "vuetify/lib/components";
```
Comment (dheepak-aot) [inline sources/packages/web/src/plugins/vuetify.ts:9]: ❤️

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @ann-aot

File: sources/packages/web/src/views/aest/institution/ViewPendingOfferingChangeRequests.vue [hunk 5]
```diff
@@ -27,27 +27,26 @@
     <content-group>
       <toggle-content :toggled="!applications?.count">
         <v-data-table-server
+          v-if="applications?.count"
           :headers="AllInProgressOfferingChangeSummaryHeaders"
           :items="applications?.results"
-          :items-length="applications?.count"
+          :items-length="applications.count"
```
Comment (dheepak-aot) [inline sources/packages/web/src/views/aest/institution/ViewPendingOfferingChangeRequests.vue:33]: 👍

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for taking care of the issues around the table. Looks good 👍

**Review by dheepak-aot — APPROVED** [review]
Looks good. 👍

**Review by guru-aot — APPROVED** [review]
LGTM, @ann-aot

**Review by andrepestana-aot — APPROVED** [review]

**Pull request merged**